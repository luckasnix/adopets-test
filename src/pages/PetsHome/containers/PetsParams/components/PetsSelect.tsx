import React from 'react'

interface Select {
  id: string;
  label: string;
  value: string;
  changed: (evt: React.ChangeEvent<HTMLSelectElement>) => void;
  options: { value: string; displayedValue: string; }[];
}

const PetsSelect: React.FunctionComponent<Select> = (props) => {
  return (
    <div>
      <label htmlFor={props.id}>{props.label}</label>
      <select id={props.id} value={props.value} onChange={props.changed}>
        {props.options.map((cur) => {
          return (
            <option key={cur.value} value={cur.value}>
              {cur.displayedValue}
            </option>
          )
        })}
      </select>
    </div>
  )
}

export default PetsSelect