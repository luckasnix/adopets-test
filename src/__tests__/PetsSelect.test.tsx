import React from 'react'
import PetsSelect from '../pages/PetsHome/containers/PetsParams/components/PetsSelect'
import renderer from 'react-test-renderer'

// test fail

// it(
//   'renders correctly',
//   () => {
//     const tree = renderer.create(<PetsSelect/>).toJSON()
//     expect(tree).toMatchSnapshot()
//   }
// )