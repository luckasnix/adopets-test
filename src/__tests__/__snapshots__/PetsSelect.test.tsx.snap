// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="input"
>
  <label
    htmlFor="sex-select"
  >
    Sex: 
  </label>
  <select
    id="sex-select"
    onChange={[Function]}
    value="sex"
  >
    <option
      value="ALL"
    >
      All
    </option>
    <option
      value="MALE"
    >
      Male
    </option>
    <option
      value="FEMALE"
    >
      Female
    </option>
  </select>
</div>
`;
